src/backend/input.rs

```diff
@@ -77,7 +77,17 @@ pub trait Event {
     fn time(&self) -> u32;
 }
 
-impl Event for () {
+/// Struct to mark events never emitted by an `InputBackend` implementation.
+///
+/// Implements all event types and can be used in place for any `Event` type,
+/// that is not used by an `InputBackend` implementation. Initialization is not
+/// possible, making accidential use impossible and enabling a lot of possible
+/// compiler optimizations.
+pub struct UnusedEvent {
+    _hidden_field: (),
+}
```

Comment:
Just use `pub enum UnusedEvent {};`, an empty enum. This would be statically guaranteed to never be instanciable, and rustc understands this.

Also, all the impls of methods for this type would become:

```rust
fn foo(&self) -> ... {
    match *self {}
}